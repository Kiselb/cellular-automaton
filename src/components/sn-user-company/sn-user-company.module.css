.toggle {
    font-size: 1em;
    text-decoration: underline;
    cursor: pointer;
}
.company {
    margin-left: 2em;
}
