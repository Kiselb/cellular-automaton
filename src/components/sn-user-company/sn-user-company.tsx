import React, { Component } from "react";
import { TUserCompany } from "../sn-user/sn-user";
import styles from "./sn-user-company.module.css";

type TState = {
  display: boolean;
};
export class UserCompany extends Component<TUserCompany, TState> {
  constructor(props: TUserCompany) {
    super(props);
    this.state = {
      display: false,
    };
    this.toggleCompany = this.toggleCompany.bind(this);
  }
  toggleCompany() {
    this.setState((prevState) => ({ display: !prevState.display }));
  }

  render() {
    return (
      <div data-testid="1" onClick={this.toggleCompany}>
        <div className={styles.toggle}>
          {!this.state.display ? "View Company" : "Hide Company"}
        </div>
        {this.state.display ? (
          <div className={styles.company}>
            <div>{this.props.name}</div>
            <div>{this.props.catchPhrase}</div>
            <div>{this.props.bs}</div>
          </div>
        ) : null}
      </div>
    );
  }
}
