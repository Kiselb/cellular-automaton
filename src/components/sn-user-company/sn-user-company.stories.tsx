import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";

import { UserCompany } from "./sn-user-company";

export default {
  title: "Messenger/User",
  component: UserCompany,
  argTypes: {
    name: { control: "string" },
    catchPhrase: { control: "string" },
    bs: { control: "string" },
  },
} as ComponentMeta<typeof UserCompany>;

const Template: ComponentStory<typeof UserCompany> = (args) => (
  <UserCompany {...args} />
);

export const AliceCompany = Template.bind({});
AliceCompany.args = {
  name: "Company Name",
  catchPhrase: "catchPhrase",
  bs: "bs",
};
