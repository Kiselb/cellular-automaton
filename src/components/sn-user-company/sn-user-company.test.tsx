import React from "react";
import { render, fireEvent, screen } from "@testing-library/react";

import "@testing-library/jest-dom";
import { UserCompany } from "./sn-user-company";

it("Company renders correctly", () => {
  const company = {
    name: "Romaguera-Crona",
    catchPhrase: "Multi-layered client-server neural-net",
    bs: "harness real-time e-markets",
  };

  const { asFragment } = render(<UserCompany {...company} />);
  screen.debug();

  expect(asFragment()).toMatchSnapshot();
});
it("Company Hide", () => {
  const company = {
    name: "Romaguera-Crona",
    catchPhrase: "Multi-layered client-server neural-net",
    bs: "harness real-time e-markets",
  };

  const { asFragment } = render(<UserCompany {...company} />);
  const div = screen.getByTestId("1");
  fireEvent.click(div);

  screen.debug();

  expect(asFragment()).toMatchSnapshot();
});
