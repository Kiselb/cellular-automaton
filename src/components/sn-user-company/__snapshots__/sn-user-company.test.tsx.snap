// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Company Hide 1`] = `
<DocumentFragment>
  <div
    data-testid="1"
  >
    <div>
      Hide Company
    </div>
    <div>
      <div>
        Romaguera-Crona
      </div>
      <div>
        Multi-layered client-server neural-net
      </div>
      <div>
        harness real-time e-markets
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`Company renders correctly 1`] = `
<DocumentFragment>
  <div
    data-testid="1"
  >
    <div>
      View Company
    </div>
  </div>
</DocumentFragment>
`;
